packages/user-account/src/useCases/user.service.js

```diff
@@ -0,0 +1,15 @@
+const boom = require('@hapi/boom');
+const validatorHandler = require('../utils/validator/validatorHandler');
+const { createUserSchema } = require('../utils/validator/schema/user.schema');
+
+function createUser(model) {
+  const data = validatorHandler(createUserSchema, model);
+  if (data !== model) {
+    return boom.badRequest(data).details;
```

Comment:
I understand the idea of validating inputs in the use case, but how do you know when the returned result was from a success operation or not?

For example, I was trying to call your endpoint and this was the response:
![Screen Shot 2022-04-15 at 12 28 13](https://user-images.githubusercontent.com/28733681/163624031-ade0d353-b694-4460-a642-c17d69303ade.png)

As you can see, it is returning the error en the same variable as the success result, and the status code is 200... it's supposed to return 400 in this scenario but, how do you know which status should be returned?

Reply:
in this case since the data format is part of the business rules it must be checked by the service and also with Joi it filters more specific data types such as an email string or a date string.

However, I am still investigating how to pass the error to fastify.

Reply:
I have already applied the settings to give a return 400 in which the service does the following